### PR TITLE
Where and post filter

### PR DESCRIPTION
**Pull request by lorenzo**
* Tests and documentation for where()
* Implemented postFilter()

### CODE REVIEW

File: src/Query.php [hunk 1]
```diff
@@ -154,19 +154,92 @@ public function order($order, $overwrite = false)
...
+    * @param array|callable|Elastica\Filter\AbstractFilter $conditions The list of conditions.
+    * @param bool $overwrite Whether or not to replace previous filters.
+    * @return $this
+    * @see Cake\ElasticSearch\FilterBuilder
+    */
     public function where($conditions, $overwrite = false)
     {
-        if ($this->_parts['preFilter'] === null || $overwrite) {
-            $this->_parts['preFilter'] = new BoolFilter;
+        return $this->_buildFilter('preFilter', $conditions, $overwrite);
```
Comment (jippi) [inline src/Query.php]: afaik in ES its called plain "filter" without the "pre"

Comment (markstory) [inline src/Query.php]: I think preFilter is an elastica thing.

**Comment by markstory** [comment]
Nice work @lorenzo

File: src/Query.php (hunk 1 above)
Comment (lorenzo) [inline src/Query.php]: preFilter is something I made up, it is not going to elastica or to elastic search

Comment (markstory) [inline src/Query.php]: Oh ok.

**Pull request merged**